sentry-android-core/src/test/java/io/sentry/android/core/ConnectivityCheckerTest.kt

```diff
@@ -1,31 +1,161 @@
 package io.sentry.android.core
 
 import android.content.Context
-import androidx.test.core.app.ApplicationProvider
-import androidx.test.ext.junit.runners.AndroidJUnit4
+import android.content.pm.PackageManager.PERMISSION_DENIED
+import android.net.ConnectivityManager
+import android.net.Network
+import android.net.NetworkCapabilities
+import android.net.NetworkCapabilities.TRANSPORT_CELLULAR
+import android.net.NetworkCapabilities.TRANSPORT_ETHERNET
+import android.net.NetworkCapabilities.TRANSPORT_WIFI
+import android.net.NetworkInfo
+import android.os.Build
+import com.nhaarman.mockitokotlin2.any
+import com.nhaarman.mockitokotlin2.eq
 import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.whenever
 import io.sentry.android.core.util.ConnectivityChecker
-import kotlin.test.BeforeTest
 import kotlin.test.Test
-import org.junit.runner.RunWith
+import kotlin.test.assertEquals
+import kotlin.test.assertFalse
+import kotlin.test.assertNull
+import kotlin.test.assertTrue
 
-@RunWith(AndroidJUnit4::class)
 class ConnectivityCheckerTest {
 
-    private lateinit var context: Context
+    @Test
+    fun `When network is active and connected, return true for isConnected`() {
+        val contextMock = mock<Context>()
+        val cm = mock<ConnectivityManager>()
+        whenever(contextMock.getSystemService(any())).thenReturn(cm)
+        val an = mock<NetworkInfo>()
+        whenever(cm.activeNetworkInfo).thenReturn(an)
+        whenever(an.isConnected).thenReturn(true)
+        assertTrue(ConnectivityChecker.isConnected(contextMock, mock())!!)
+    }
+
+    @Test
+    fun `When network is active but not connected, return false for isConnected`() {
+        val contextMock = mock<Context>()
+        val cm = mock<ConnectivityManager>()
+        whenever(contextMock.getSystemService(any())).thenReturn(cm)
+        val an = mock<NetworkInfo>()
+        whenever(cm.activeNetworkInfo).thenReturn(an)
+        whenever(an.isConnected).thenReturn(false)
+        assertFalse(ConnectivityChecker.isConnected(contextMock, mock())!!)
+    }
+
+    @Test
+    fun `When there's no permission, return true for isConnected`() {
+        val contextMock = mock<Context>()
+        val cm = mock<ConnectivityManager>()
+        whenever(contextMock.getSystemService(any())).thenReturn(cm)
+        whenever(contextMock.checkPermission(any(), any(), any())).thenReturn(PERMISSION_DENIED)
+        assertTrue(ConnectivityChecker.isConnected(contextMock, mock())!!)
```

Comment:
`m`: Wouldn't it be better to return `null` or something else instead? Is it fine to assume we have an internet connection? If we keep it like this, I would definitely add this as a comment to `isConnected`.

Reply:
we still want to be able to send events over the network if no permission is granted to check the network status, the connection can be active or not, we dont know.